test/sqllogictest/arithmetic.slt

```diff
@@ -519,3 +519,121 @@ SELECT 9223372036854775807::bigint+1::bigint
 
 query error numeric field overflow
 SELECT 9223372036854775807::bigint-(-1)::bigint
+
+### bitwise operators ###
+
+# negation
+
+query IIIII
+SELECT ~0, ~1, ~65535, ~2147483647, ~9223372036854775807
+----
+-1 -2 -65536 -2147483648 -9223372036854775808
+
+# bitwise and
+
+query IIIII
+SELECT
+    3 & 4,
+    7 & 4,
+    7 & 14,
+    1342579021 & 1035032143,
+    2309456347094635094 & 4321056356463245221
+----
+0 4 6 268435533 2307057198754714116
+
+# bitwise or
+
+query IIIII
+SELECT
+    3 | 4,
+    7 | 4,
+    7 | 14,
+    1342579021 | 1035032143,
+    2309456347094635094 | 4321056356463245221
+----
+7 7 15 2109175631 4323455504803166199
+
+# bitwise xor
+
+query IIIII
+SELECT
+    3 # 4,
+    7 # 4,
+    7 # 14,
+    1342579021 # 1035032143,
+    2309456347094635094 # 4321056356463245221
+----
+7 3 9 1840740098 2016398306048452083
+
+
+# precedence between '&' and '|'
```

Comment:
Thank you for the tests! You can also display/make explicit the precedence using an EXPLAIN statement of a SELECT that uses actual columns (so the expression are not constant-folded). The EXPLAIN should show the order of operations using nested brackets on the `Map` line:
```
materialize=> explain select f1 + f1 - f1 from t1;
           Optimized Plan            
-------------------------------------
 %0 =                               +
 | Get materialize.public.t1 (u1667)+
 | Map ((#0 + #0) - #0)             +
 | Project (#1)                     +
```

Reply:
Didn't think of that, good point!